packages/core/ui-settings/core-ui-settings-browser/src/types.ts

```diff
@@ -100,6 +100,11 @@ export interface IUiSettingsClient {
    * the settings, containing the actual Error class.
    */
   getUpdateErrors$: () => Observable<Error>;
+
+  /**
+   * Returns a validation error message if the uiSettings value is invalid.
+   */
+  validateValue: (key: string, value: any) => Promise<string | null>;
```

Comment:
Yeah, that makes sense to me, especially for the server-side response, where it would be weird to return an empty response body when the value is valid. 

For the browser-side API, I thought it would be helpful to also return an additional boolean to indicate whether the validate request was successful:
```ts
type ValueValidation = { successfulValidation: boolean, valid?: boolean, errorMessage?: string };

validateValue: (key: string, value: any) => Promise<ValueValidation>;
```
The other option I considered was returning `Promise<ValueValidation | null>` so that we return `null` if we catch any errors, but I thought the pattern with `successfulValidation` might be more convenient to use on the client-side. WDYT?